### PR TITLE
Run unit tests in Github Actions

### PR DESCRIPTION
**Pull request by krisgesling**
**Description:**
Adds automated testing for any new PR via Github Actions.

Intentionally commented out flake8 checks at the moment as these will fail on the current codebase.

**How to test:**
See automated checks below.

**CLA**
- [x] Yes

### CODE REVIEW

File: .github/workflows/python-app.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install flake8 pytest
+        python setup.py install
+    # - name: Lint with flake8
+    #   run: |
+    #     # stop the build if there are Python syntax errors or undefined names
+    #     flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
```
Comment (forslund) [inline .github/workflows/python-app.yml:34]: I think this flake8 command can be run to verify some basic mistakes (usage of non-existing variables etc)...while the one below can be commented until the code style of this project is decided upon.

**Review by forslund — COMMENTED** [review]
Looks good and is working, did comment on the flake-8 test which can be partially activated to check for some common mistakes.

File: .github/workflows/python-app.yml (hunk 1 above)
Comment (krisgesling) [reply .github/workflows/python-app.yml]: Fair call, I've also added an inline comment detailing why the styling check is commented out

File: .github/workflows/python-app.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        python-version: [3.6, 3.7, 3.8, 3.9]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
```
Comment (clusterfudge) [inline .github/workflows/python-app.yml:27]: I think I would prefer that these be pulled out into a script. This allows developers to run them as part of their developer workflow and easily reproduce what's happening on the build action.

File: .github/workflows/python-app.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        python -m pip install --upgrade pip
+        python -m pip install flake8 pytest
+        python setup.py install
+    - name: Lint with flake8
```
Comment (clusterfudge) [inline .github/workflows/python-app.yml:31]: same as above.

**Review by clusterfudge — CHANGES_REQUESTED** [review]

**Comment by forslund** [comment]
Discussing abit with @krisgesling yesterday and we've now updated to use test-requirements.txt to install pytest and flake8.

The run_tests.py script is removed for the time being since it's not used anymore. testing instructions have been added to the readme for now. A new script should probably be created to allow running the flake8 and pytest...

File: README.md [hunk 4]
```diff
@@ -86,6 +86,26 @@ def register_pandora_vocab(emitter):
...
+```
+pip install -r test-requirements.txt
+```
+
+This will install the test-requirements as well as the runtime requirements for adapt.
+
+To test any changes before submitting them run
+
+```
+pytest test/*
```
Comment (clusterfudge) [inline README.md:106]: I think that we should include the flake8 command in the instructions (as that will also fail the remote build), and we should pull both into a single script for simpler instructions.

**Comment by forslund** [comment]
Since the flake command-line is non-trivial I pulled it out into a `run_tests.sh` script as you suggested @clusterfudge, readme has been updated with this information.

**Review by clusterfudge — APPROVED** [review]
lgtm!

Thanks @forslund for pulling this across the line. Please open an issue for us to address/document&ignore the lint errors we're excluding with `--select=E9,F63,F7,F82`

**Comment by forslund** [comment]
Will write up an issue during the day.

**Pull request merged**